client/src/pages/Courses/RegulationModal.js

```diff
@@ -93,27 +154,28 @@ const RegulationModal = ({ isOpen, onClose, updateRegs, title }) => {
                     <ModalBody>
                         <FormControl as="fieldset">
                             <Flex align="center">
-                                <Switch
-                                    id="course-note-switch"
-                                    m="2"
-                                    onChange={() => setToggleInput(!toggleInput)}
-                                    isChecked={toggleInput}
-                                />
+                                <Switch id="course-note-switch" m="2" onChange={handleToggleChange} isChecked={toggleInput} />
                                 <FormLabel m="2">{toggleInput ? "Add course regulation" : "Add note"}</FormLabel>
                             </Flex>
                             {toggleInput ? (
                                 <>
                                     <Flex m="2">
                                         <Input id="courses" placeholder="Courses" onChange={handleCourseChange} value={courseTextValue} />
-                                        <IconButton aria-label="Add input" icon="arrow-down" onClick={handleCourseClick} />
+                                        <IconButton
+                                            aria-label="Add input"
+                                            icon="arrow-down"
+                                            onClick={handleCourseClick}
+                                            isDisabled={courseTextValue !== "" ? false : true}
```

Comment:
isDisabled={courseTextValue === "" }

Reply:
Alternatively `isDisabled={!courseTextValue}` since empty string is falsey in JS

Reply:
^ use this. reads better

Reply:
Fixed